tasks/Task.cpp

```diff
@@ -269,20 +269,36 @@ void Task::processRawInputs()
     for (auto& binding : m_bound_raw_in) {
         GstElement* appsrc = binding.app_element;
         while (binding.port->read(binding.memory, false) == RTT::NewData) {
-            pushRawData(*appsrc, binding.memory.data);
+            GstClockTime pipeline_running_time{
+                gst_element_get_current_running_time(m_pipeline)};
+            // live sources must be timestamped with the pipeline running time. See
+            // gstreamer.freedesktop.org/documentation/additional/design/live-source.html?gi-language=c#timestamps
+            pushRawData(*appsrc, binding.memory.data, pipeline_running_time);
         }
     }
 }
 
-void Task::pushRawData(GstElement& appsrc, vector<std::uint8_t> const& data)
+void Task::pushRawData(GstElement& appsrc,
+    vector<std::uint8_t> const& data,
+    GstClockTime const& timestamp)
 {
-    GstUnrefGuard buffer(gst_buffer_new_and_alloc(data.size()));
-    gst_buffer_fill(buffer.get(), 0, data.data(), data.size());
-    GstFlowReturn ret;
-    g_signal_emit_by_name(&appsrc, "push-buffer", buffer.get(), &ret);
+    size_t pushed{0};
+    while (pushed < data.size()) {
+        GstUnrefGuard buffer(gst_buffer_new_and_alloc(data.size()));
+        if (!buffer.get()) {
+            continue;
+        }
+
+        size_t n =
```

Comment:
I don't understand how `n` can be anything else than `data.size()` here. 

<img width="728" height="44" alt="Image" src="https://github.com/user-attachments/assets/76b1ca80-0d80-4836-b855-21a86bb8b060" />

In our case, the buffer is initialized to be `data.size()` bytes long, so that does not apply IMO.